### PR TITLE
[ModelIO] Adds Xcode 9 Beta 1 to Stable API bindings

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

File: src/ModelIO/MDLAnimatedValueTypes.cs [hunk 1]
```diff
@@ -0,0 +1,693 @@
...
+		{
+			if (values == null)
+				throw new ArgumentNullException (nameof (values));
+			if (times == null)
+				throw new ArgumentNullException (nameof (times));
+
+			unsafe {
+				fixed (Vector3d* valuesPtr = values)
+				fixed (double* timesPtr = times)
+					_ResetWithDouble3Array ((IntPtr) valuesPtr, (IntPtr) timesPtr, (nuint) times.Length);
```
Comment (rolfbjarne) [inline src/ModelIO/MDLAnimatedValueTypes.cs]: What's the length of the `values` array supposed to be? Identical to the length of `times`? If so, we should add checks for it.

File: src/modelio.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 //
 // Copyright 2015 Xamarin, Inc.
 //
-//
+//
```
Comment (rolfbjarne) [inline src/modelio.cs]: Minor: whitespace noise.

File: tests/monotouch-test/ModelIO/MDLAnimatedValueTypesTests.cs [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+using MonoTouch.ObjCRuntime;
+#endif
+using OpenTK;
+using Bindings.Test;
+using NUnit.Framework;
+
+namespace MonoTouchFixtures.ModelIO {
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class A_MDLAnimatedValueTypesTests {
```
Comment (rolfbjarne) [inline tests/monotouch-test/ModelIO/MDLAnimatedValueTypesTests.cs]: Please remove the `A_` prefix :smile:

File: src/modelio.cs [hunk 4]
```diff
@@ -1571,12 +1667,26 @@ interface MDLTexture : MDLNamed
...
+		bool WriteToUrl (NSUrl nsurl, string type, nuint level);
+
+		[Export ("imageFromTexture")]
+		[return: NullAllowed]
 		CGImage GetImageFromTexture ();
 
+		[TV (11,0), Mac (10,13, onlyOn64:true), iOS (11,0)]
+		[Export ("imageFromTextureAtLevel:")]
+		[return: NullAllowed]
+		CGImage GetImageFromTexture (nuint atLevel);
```
Comment (VincentDondain) [inline src/modelio.cs]: It's kinda weird to name it `atLevel`. `level` should be good enough.

You then do `atLevel = 3`...

File: src/modelio.cs [hunk 5]
```diff
@@ -2234,5 +2345,655 @@ interface MDLSubmeshTopology {
...
+		[Abstract]
+		[Export ("IsInverseOp")]
+		bool IsInverseOp { get; }
+	}
+
+	[iOS (11,0), Mac (10,13, onlyOn64 : true), TV (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MDLTransformRotateXOp : MDLTransformOp {
+
+		// From MDLTransformOp Protocol
```
Comment (VincentDondain) [inline src/modelio.cs]: Why do we want to keep this commented code around?

**Pull request closed without merging**